app/main.py

```diff
@@ -1,2 +1,102 @@
+from typing import Any, Iterator
+from dataclasses import dataclass
+
+
+@dataclass
+class Node:
+    key: Any
+    _hash: int
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.hash_table: list = [None] * self.capacity
+
+    def __len__(self) -> int:
+        return self.length
+
+    def _hash(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def find_index(self, key: Any) -> int:
+        index = self._hash(key)
+
+        while True:
+            node = self.hash_table[index]
+            if node is None:
+                return index
+            if node.key == key:
+                return index
+            index = (index + 1) % self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.length + 1 >= self.capacity * self.load_factor:
+            self.resize_dictionary()
+
+        index = self.find_index(key)
+        if self.hash_table[index] is None:
+            self.hash_table[index] = Node(key, hash(key), value)
+            self.length += 1
+        else:
+            self.hash_table[index].value = value
+
+    def resize_dictionary(self) -> None:
+        old_hash_table = [item for item in self.hash_table if item]
+        self.hash_table = (len(self.hash_table) * 2) * [None]
+        self.capacity *= 2
+        self.length = 0
+        for item in old_hash_table:
+            self.__setitem__(item.key, item.value)
+
+    def check_valid_index(self, key: Any) -> int:
+        index = self._hash(key)
+        start_index = index
+        while self.hash_table[index]:
+            if self.hash_table[index].key == key:
+                return index
+            index = (index + 1) % self.capacity
+            if start_index == index:
+                raise KeyError("Key not found in dictionary")
+        return index
+
+    def __getitem__(self, key: Any) -> None:
```

Comment:
The return type annotation for `__getitem__` should be `Any` instead of `None`, as this method is expected to return the value associated with the given key.